src/rtctools/data/pi.py

```diff
@@ -454,7 +455,7 @@ def __init__(
                     )
 
                 el = header.find("pi:forecastDate", ns)
-                if el is not None:
+                if el is not None and pi_read_forecast_date:
                     forecast_datetime = self.__parse_date_time(el)
                 else:
                     # the timeseries has no forecastDate, so the forecastDate
```

Comment:
Doesn't this mean that you will now just get an error if all start datetimes don't match? That doesn't seem like much of an improvement